Add brief feature description (#298)
rancher/turtles-docs: docs/next/modules/en/pages/reference/features.adoc

| alpha
|===

=== Day 2 Operations

> This feature gate is used to enable or disable all subfeatures that are typically used for the ongoing management and maintenance of your Kubernetes clusters, after they have been provisioned. Currently, there is a single subfeature available for RKE2 clusters that allows for etcd snapshots to be created and restored in downstream clusters, from the management cluster. In order to enable this subfeature, you need to enable both the `day2operations` and the `etcdBackupRestore` feature gates by setting the relevant values of the Turtles Helm chart. Note that this feature is currently in alpha stage with no long-term support.

Reviewer: ```suggestion
This feature gate enables or disables all subfeatures typically used for the ongoing management and maintenance of your Kubernetes clusters after they have been provisioned. Currently, a single subfeature is available for RKE2 clusters, allowing etcd snapshots to be created and restored in downstream clusters from the management cluster. To enable this subfeature, you must enable both the `day2operations` and `etcdBackupRestore` feature gates by setting the relevant values in the Turtles Helm chart. Note that this feature is currently in alpha and does not have long-term support.
```

=== Add-on Provider for Fleet

> This feature installs Cluster API Add-on Provider for Fleet (CAAPF) to provide integration with link:https://github.com/rancher/fleet[Fleet] and enable the easy deployment of applications to CAPI provisioned clusters. CAAPF automatically installs Fleet in the management cluster and registers newly provisioned CAPI clusters with Fleet, so that applications can be deployed to these clusters via GitOps, `Bundle` or `HelmApp` resources. CAAPF also includes support for grouping CAPI clusters that reference the same ClusterClass through Fleet's ClusterGroup, essentially simplifying application deployments to large groups of clusters. This feature is in GA stage and enabled by default.

Reviewer: ```suggestion
This feature installs the Cluster API Add-on Provider for Fleet (CAAPF) to integrate with Fleet and enable the easy deployment of applications to CAPI-provisioned clusters. CAAPF automatically installs Fleet in the management cluster and registers newly provisioned CAPI clusters with Fleet, allowing applications to be deployed to these clusters via GitOps, Bundle, or HelmApp resources. CAAPF also supports grouping CAPI clusters that reference the same ClusterClass using Fleet’s ClusterGroup, simplifying application deployments to large groups of clusters. This feature is generally available (GA) and enabled by default.
```

=== Agent TLS Mode

> This feature gate is used to instruct Turtles on whether it should follow Rancher's `agent-tls-mode` setting, when retrieving cluster registration manifests for importing clusters. When enabled, Turtles will lookup the value of Rancher's `agent-tls-mode` setting to determine whether it has been set to `system-store` or `strict`. If set to `system-store`, Turtles will trust any certificate generated by a public Certificate Authority contained in the operating system's trust store, including those signed by authorities such as Let's Encrypt. If set to `strict`, Turtles will only trust certificates generated by the Certificate Authority contained in Rancher's `cacerts` setting. When this setting is disabled, Turtles will fallback to work in `system-store` mode. This feature is in beta stage and enabled by default.

Reviewer: ```suggestion
This feature gate instructs Turtles whether to follow Rancher’s agent-tls-mode setting when retrieving cluster registration manifests for importing clusters. When enabled, Turtles looks up Rancher’s agent-tls-mode setting to determine if it is set to system-store or strict. If set to system-store, Turtles will trust any certificate issued by a public Certificate Authority included in the operating system’s trust store, including those signed by authorities such as Let’s Encrypt. If set to strict, Turtles will trust only certificates issued by the Certificate Authority specified in Rancher’s cacerts setting. When this setting is disabled, Turtles defaults to system-store mode. This feature is in beta and enabled by default.
```

=== Cluster Class Operations

> This feature gate is used to enable operations against CAPI clusters that use the ClusterClass feature. As CAPI clusters evolve, there is often a need to transition them from one ClusterClass to another, an operation known as rebasing. Turtles includes experimental support for rebasing which can be tested by enabling this feature gate.  Note that this feature is currently in alpha stage with no long-term support.

Reviewer: ```suggestion
This feature gate enables operations on CAPI clusters that use the ClusterClass feature. As CAPI clusters evolve, it is often necessary to transition them from one ClusterClass to another—an operation known as rebasing. Turtles includes experimental support for rebasing, which can be tested by enabling this feature gate. Note that this feature is currently in alpha and does not have long-term support.
```
